raw-window-handle implementation broken on macOS

As pointed out by @SamuelMarks in [#2213](https://github.com/linebender/druid/pull/2213#issuecomment-1174017123) `druid-shell` fails to compile right now when the `raw-win-handle` feature in enabled.

```
error[E0432]: unresolved import `raw_window_handle::macos`
  --> druid-shell/src/backend/mac/window.rs:43:25
   |
43 | use raw_window_handle::{macos::MacOSHandle, HasRawWindowHandle, RawWindowHandle};
   |                         ^^^^^ could not find `macos` in `raw_window_handle`

error[E0432]: unresolved import `crate::dialog::FileDialogType`
  --> druid-shell/src/backend/mac/dialog.rs:27:40
   |
27 | use crate::dialog::{FileDialogOptions, FileDialogType};
   |                                        ^^^^^^^^^^^^^^ no `FileDialogType` in `dialog`

error[E0432]: unresolved import `crate::dialog::FileDialogType`
  --> druid-shell/src/backend/mac/window.rs:58:40
   |
58 | use crate::dialog::{FileDialogOptions, FileDialogType};
   |                                        ^^^^^^^^^^^^^^ no `FileDialogType` in `dialog`

error[E0599]: no variant or associated item named `MacOS` found for enum `RawWindowHandle` in the current scope
    --> druid-shell/src/backend/mac/window.rs:1422:26
     |
1422 |         RawWindowHandle::MacOS(handle)
     |                          ^^^^^ variant or associated item not found in `RawWindowHandle`
```

Probably related to [the breaking change](https://github.com/rust-windowing/raw-window-handle/blob/master/CHANGELOG.md#040-2021-11-15) in `raw-window-handle` version `0.4.0` that renamed `MacOS` to `AppKit`. Does our code even specify semver related compatibility with `raw-window-handle`? Can't see it after a quick skim.

Needs further investigation..

--- a/druid-shell/src/backend/mac/window.rs
+++ b/druid-shell/src/backend/mac/window.rs
@@ -40,7 +40,7 @@ use objc::{class, msg_send, sel, sel_impl};
 use tracing::{debug, error, info};
 
 #[cfg(feature = "raw-win-handle")]
-use raw_window_handle::{macos::MacOSHandle, HasRawWindowHandle, RawWindowHandle};
+use raw_window_handle::{AppKitHandle, HasRawWindowHandle, RawWindowHandle};
 
 use crate::kurbo::{Insets, Point, Rect, Size, Vec2};
 use crate::piet::{Piet, PietText, RenderContext};
@@ -1417,11 +1417,9 @@ impl WindowHandle {
 unsafe impl HasRawWindowHandle for WindowHandle {
     fn raw_window_handle(&self) -> RawWindowHandle {
         let nsv = self.nsview.load();
-        let handle = MacOSHandle {
-            ns_view: *nsv as *mut _,
-            ..MacOSHandle::empty()
-        };
-        RawWindowHandle::MacOS(handle)
+        let mut handle = AppKitHandle::empty();
+        handle.ns_view = *nsv as *mut _;
+        RawWindowHandle::AppKit(handle)
     }
 }
 

